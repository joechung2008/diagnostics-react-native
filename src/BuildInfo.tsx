import React from 'react';
import { View, Text } from 'react-native';
import { useTheme } from './ThemeContext';

const BuildInfo: React.FC<BuildInfoProps> = ({ buildVersion }) => {
  const { colors } = useTheme();

  const items = [
    {
      name: 'Build Version',
      value: buildVersion,
    },
  ];

  const dynamicStyles = {
    container: {
      padding: 10,
      backgroundColor: colors.background,
    },
    table: {
      borderWidth: 1,
      borderColor: colors.tableBorder,
      backgroundColor: colors.surface,
    },
    header: {
      flexDirection: 'row' as const,
      borderBottomWidth: 1,
      borderBottomColor: colors.tableBorder,
      paddingTop: 5,
      paddingBottom: 5,
      backgroundColor: colors.tableHeaderBg,
    },
    headerText: {
      flex: 1,
      fontWeight: 'bold' as const,
      paddingLeft: 5,
      paddingRight: 5,
      color: colors.text,
    },
    row: {
      flexDirection: 'row' as const,
      paddingTop: 5,
      paddingBottom: 5,
      borderBottomWidth: 1,
      borderBottomColor: colors.tableBorder,
    },
    cell: {
      flex: 1,
      paddingLeft: 5,
      paddingRight: 5,
      minWidth: 0, // Allow flex items to shrink below their content size
    },
    cellText: {
      flex: 1,
      flexShrink: 1,
      color: colors.text,
    },
  };

  return (
    <View style={dynamicStyles.container}>
      <View style={dynamicStyles.table}>
        <View style={dynamicStyles.header}>
          <Text style={dynamicStyles.headerText}>Name</Text>
          <Text style={dynamicStyles.headerText}>Value</Text>
        </View>
        {items.map(item => (
          <View key={item.name} style={dynamicStyles.row}>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.name}
            </Text>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.value}
            </Text>
          </View>
        ))}
      </View>
    </View>
  );
};

export default BuildInfo;
