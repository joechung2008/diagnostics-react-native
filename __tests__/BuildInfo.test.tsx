import BuildInfo from '@/BuildInfo';
import { render, screen } from '@testing-library/react-native';
import React from 'react';

// Mock the ThemeContext
jest.mock('@/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f8f9fa',
      text: '#212529',
      tableBorder: '#dee2e6',
      tableHeaderBg: '#f8f9fa',
    },
  }),
}));

describe('BuildInfo', () => {
  it('renders correctly with build version', () => {
    render(<BuildInfo buildVersion="1.0.0" />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
