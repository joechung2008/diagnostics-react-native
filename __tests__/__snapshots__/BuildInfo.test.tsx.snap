// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BuildInfo renders correctly with build version 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f8f9fa",
        "borderColor": "#dee2e6",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#f8f9fa",
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Build Version
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        1.0.0
      </Text>
    </View>
  </View>
</View>
`;
